File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/CPP/PlusOne.cpp
Given a number represented as an array of digits, plus one to the number.

// Question Description above

class Solution {
public:
    vector<int> plusOne(vector<int> &digits) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int ten;
        vector<int>::iterator it = digits.end();
        it --;
        ten = 1;
        while ( it != digits.begin() && ten == 1 ){
            ten = ( ten + (*it) ) / 10;
            *it = ( 1 + (*it) ) % 10;
            it --;
        }
        if( ten == 1 ){
            ten = ( ten + (*it) ) / 10;
            *it = ( 1 + (*it) ) % 10;
        }
        if( ten == 1 )
            digits.insert( digits.begin(), 1 );
        return digits;
    }
};

File: SQL-AI/ProgrammingLanguage/Leetcode-Implementation/Python/PlusOne.py
class Solution:
    # @param digits, a list of integer digits
    # @return a list of integer digits
    def plusOne(self, digits):
        if len(digits) == 0:    return [1]
        
        pos = len(digits) - 1
        carry = 1
        while pos >= 0 and carry == 1:
            temp = digits[pos] + carry
            carry = int(temp /10 )
            digits[pos] = temp%10
            pos -= 1
        if carry == 1:
            digits.insert(0, carry)
        return digits 
